freqtrade/enums/interestmode.py

```diff
@@ -0,0 +1,30 @@
+from decimal import Decimal
+from enum import Enum
+from math import ceil
+
+from freqtrade.exceptions import OperationalException
+
+
+one = Decimal(1.0)
+four = Decimal(4.0)
+twenty_four = Decimal(24.0)
+
+
+class InterestMode(Enum):
+    """Equations to calculate interest"""
+
+    HOURSPERDAY = "HOURSPERDAY"
+    HOURSPER4 = "HOURSPER4"  # Hours per 4 hour segment
+    NONE = "NONE"
+
+    def __call__(self, *args, **kwargs):
```

Comment:
If it doesn't work then couldn't we just convert the string to the enum once we pull it out again? I'm not sure what to do about this, I am using sqlalchemy's enum type `interest_mode = Column(Enum(InterestMode), nullable=True)
`, and to me the fact that they have the type to begin with hints that it will be of that type when its taken out, otherwise I don't know why they would create it. It seems kind of pointless if they're just going to convert it to a string.

Reply:
Databases usually don't have a "enum" type - so sqlalchemy needs to store it as string internally anyway.

The docs says that it'll make sure only the values in the enum can be stored (basically a constraint in the database if possible/supported) - which makes having an Enum type a valid case.
Converting it back is surely possible, and should be done within the method that uses it (we should not hook into sqlalchemy logic for reading a value).

To be clear, I didn't test this - so you can surely script a quick (non-freqtrade related ? ) script testing this out against the various databases - i'm just saying that i've never seen this (which only means it's probably new) - and that the docs don't seem promising to me based on the wording.

Reply:
> Converting it back is surely possible, and should be done within the method that uses it (we should not hook into sqlalchemy logic for reading a value).

I don't really know how to implement that conversion. What methods would have to be modified to do that?

> so you can surely script a quick (non-freqtrade related ? ) script testing this out against the various databases -

I've never really used `sqlalchemy`, so it might not be that quick, but I might be able to try something

Reply:
I've done some tests and this seems to really work.
will need to investigate if we can apply this to some other columns, too.